cnd/src/swap_protocols/facade.rs

```diff
@@ -146,16 +149,70 @@ impl Facade {
         self.swarm.initiate_communication(id, swap_params).await
     }
 
-    pub async fn get_finalized_swap(&self, id: LocalSwapId) -> Option<comit::FinalizedSwap> {
-        self.swarm.get_finalized_swap(id).await
-    }
-
     pub async fn get_created_swap(
         &self,
         id: LocalSwapId,
     ) -> Option<Herc20HalightBitcoinCreateSwapParams> {
         self.swarm.get_created_swap(id).await
     }
+
+    pub async fn get_alice_herc20_halight_swap(
+        &self,
+        id: LocalSwapId,
+    ) -> anyhow::Result<Option<http_api::AliceHerc20HalightBitcoinSwap>> {
+        let alpha_ledger_state = self.herc20_states.get(&id).await?;
+        let beta_ledger_state = self.halight_states.get(&id).await?;
+
+        let (alpha_ledger_state, beta_ledger_state) = match (alpha_ledger_state, beta_ledger_state)
+        {
+            (Some(alpha_ledger_state), Some(beta_ledger_state)) => {
+                (alpha_ledger_state, beta_ledger_state)
+            }
+            _ => return Ok(None),
+        };
+
+        let herc20_params = Load::<herc20::InProgressSwap>::load(self, id).await?;
+        let halight_params = Load::<halight::InProgressSwap>::load(self, id).await?;
+
```

Comment:
This cannot really happen because the state is only inserted once the swap is finalized. 

We are aware, that this is an annoying invariant. Eventually this will be resolved once we refactor where the `Option`al state lives.